integration_test/gce/zone_picker.go

```diff
@@ -0,0 +1,61 @@
+package gce
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/smallnest/weighted"
+)
+
+// weightedRoundRobin implements a thread safe weighted round robin algorithm.
+// According to the `weighted` library, it is the same algorithm as the one
+// used by nginx to do weighted round robin.
+type weightedRoundRobin struct {
+	mutex sync.Mutex
+	// This type implements the weighted round robin algorithm. I don't know what
+	// SW stands for, maybe "smooth weighted".
+	sw *weighted.SW
+}
+
+// Next() selects the next zone to spawn a VM in.
+// It is thread safe.
+func (wrr *weightedRoundRobin) Next() string {
+	wrr.mutex.Lock()
+	defer wrr.mutex.Unlock()
+
+	return wrr.sw.Next().(string)
+}
+
+// parseZoneEnvironmentVariables looks at `weightedZones` (and `zone` if
+// needed) and extracts the zones and weights into a a weighted round robin
+// zone picker.
+// TODO(martijnvans): Remove support for `zone`.
+func newZonePicker(weightedZones, zone string) (*weightedRoundRobin, error) {
+	if weightedZones == "" {
+		if zone == "" {
+			return nil, errors.New("either ZONE or ZONES must be specified")
+		}
+		weightedZones = zone + "=1"
+	}
+
+	sw := &weighted.SW{}
+	// Each zoneSpec should look like <string>=<integer>.
+	for _, zoneSpec := range strings.Split(weightedZones, ",") {
+		zoneAndWeight := strings.Split(zoneSpec, "=")
+		if len(zoneAndWeight) != 2 {
+			return nil, fmt.Errorf(`invalid zone specification %q from ZONES=%q; should be like "us-central1=5"`, zoneSpec, weightedZones)
+		}
```

Comment:
[Optional] I suspect that in many cases, the weights will be equal, so it might make sense to support a simple comma-separated list of zones as well…

```suggestion
		if len(zoneAndWeight) == 1 {
			zoneAndWeight = append(zoneAndWeight, "1")
		} else if len(zoneAndWeight) != 2 {
			return nil, fmt.Errorf(`invalid zone specification %q from ZONES=%q; should be like "us-central1=5"`, zoneSpec, weightedZones)
		}
```

Reply:
I think supporting more than one format makes things more complicated, not simpler. Like, what do you do with this:

zone1=50,zone2

the magical =1 insertion is a surprise, I'd rather just have one consistent behavior.

Reply:
True, that one would be somewhat controversial. I was thinking more of `zone1,zone2`…

Reply:
implemented.